src/module_PI/Arduino_Uno/DataPacket.h

```diff
@@ -0,0 +1,39 @@
+/*
+  dataPacket.h - A class designed around Serial Communication
+*/
+#ifndef DataPacket_h
+#define DataPacket_h
+
+#include "Arduino.h"
+#include "../lib/Cryptosuite/Sha/sha1.h"
+
+#define SIZE_OF_HASH_IN_PACKET 8
+#define SIZE_OF_DATAPACKET_IN_BYTES 64
+
+class DataPacket{
+    public:
+      DataPacket(String message, String packetType);
+      DataPacket(byte* packetContents);
+      byte* getPacketAsArray();
+      byte* getPacketHash();
+      int getSizeInBytes();
+      String getPacketType();
+      String getMessage();
+    private:
+      void appendPacketHeader();
+      void appendHash();
+      void appendPacketTypeByte();
+      void appendMessage();
+      void append(byte rawData[], int length);
+      void append(byte singleRawByte);
+      void parseSha1FromRawPacket(byte* packetContents);
```

Comment:
One naming suggestion for these three 'parse' methods: When I read the name of them I was surprised that they weren't returning anything. I had to go and look at the method to understand what it was doing. What might make it more clear would be to name them something like "parse**AndSet**Sha...". That implies that I am parsing something out AND doing something additional with it instead of just parsing it out and returning it.

This refactor doesn't need to be included in this Pull Request and can be refactored at a future time if you decided to go that route.